python/taichi/ad/_ad.py

```diff
@@ -55,6 +64,11 @@ def __init__(self, loss=None, clear_gradients=True, validation=False):
                 Warning)
         self.eval_on_exit = loss is not None
         self.loss = loss
+        self.grad_check = grad_check
```

Comment:
Nit: IMHO it's better to put all grad_check logic in a separate class instead of putting them inside `Tape` class.